### DISCUSSION TITLE
How to find the Eigen in the same cmake project?

### DISCUSSION THREAD

**Discussion by ziyuang** [BlueBrain/HighFive#766] [Questions]
In `CMake\HighFiveTargetDeps.cmake` the relevant command for including Eigen is `target_include_directories(libdeps SYSTEM INTERFACE ${EIGEN3_INCLUDE_DIRS})`. This seems to prevent including Eigen in the same project, as doing so by `set(EIGEN3_INCLUDE_DIRS ${eigen_SOURCE_DIR})` after  [`FetchContent`](https://cmake.org/cmake/help/latest/module/FetchContent.html)ing Eigen will raise the complaint
```
 Target "libdeps" INTERFACE_INCLUDE_DIRECTORIES property contains path:

    "<project directory>/build/_deps/eigen-src"

  which is prefixed in the build directory.
```

**Comment by 1uc** [comment]
A think a similar issue is encountered in #766. A solution would be to add the HighFive sources to your project without performing the `add_subdirectory`. Then you simply reproduce the required logic yourself. See,
https://github.com/BlueBrain/libsonata/blob/f26db3226ce40a02373a9794956f23f0013de8b3/extlib/CMakeLists.txt#L2

for one example for how this could be done.
